tests/interface_dag_elements/test_results.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import numpy
+import pytest
+
+from ttsim.interface_dag_elements.results import tree
+
+
+@pytest.mark.parametrize(
+    (
+        "raw_results_columns",
+        "raw_results_params",
+        "raw_results_from_input_data",
+        "input_data__sort_indices",
+        "expected_result",
+    ),
+    [
+        (
+            # Raw results columns with sorted data
+            {"result_col": numpy.array([200, 300, 100])},
```

Comment:
Can you add test cases 
```suggestion
            {"result_col": numpy.array([200, 300, 100]), "result_scalar": 1.0},
```

Also add the same thing to `raw_results_from_input_data` and some dict or so to `raw_results_from_params`, so we check all cases that we made provisions for.